pkg/proxy/healthcheck/service_health.go

```diff
@@ -59,20 +58,18 @@ type proxierHealthChecker interface {
 }
 
 func newServiceHealthServer(hostname string, recorder events.EventRecorder, listener listener, factory httpServerFactory, nodePortAddresses *utilproxy.NodePortAddresses, healthzServer proxierHealthChecker) ServiceHealthServer {
+	var nodeIPs []net.IP
 
-	nodeAddresses, err := nodePortAddresses.GetNodeAddresses(utilproxy.RealNetwork{})
-	if err != nil || nodeAddresses.Len() == 0 {
-		klog.ErrorS(err, "Failed to get node ip address matching node port addresses, health check port will listen to all node addresses", "nodePortAddresses", nodePortAddresses)
-		nodeAddresses = sets.New[string]()
-		nodeAddresses.Insert(utilproxy.IPv4ZeroCIDR)
-	}
-
-	// if any of the addresses is zero cidr then we listen
-	// to old style :<port>
-	for _, addr := range nodeAddresses.UnsortedList() {
-		if utilproxy.IsZeroCIDR(addr) {
-			nodeAddresses = sets.New[string]("")
-			break
+	if nodePortAddresses.MatchAll() {
```

Comment:
bikeshedding :) you already made this clearer but I wonder if we still remove another indentation
```diff
-       var nodeIPs []net.IP
-
-       if nodePortAddresses.MatchAll() {
-               // It doesn't matter whether we listen on "0.0.0.0", "::", or ""; go
-               // treats them all the same.
-               nodeIPs = []net.IP{net.IPv4zero}
-       } else {
-               var err error
-               nodeIPs, err = nodePortAddresses.GetNodeIPs(utilproxy.RealNetwork{})
+       // It doesn't matter whether we listen on "0.0.0.0", "::", or ""; go
+       // treats them all the same.
+       nodeIPs := []net.IP{net.IPv4zero}
+       if !nodePortAddresses.MatchAll() {
+               ips, err := nodePortAddresses.GetNodeIPs(utilproxy.RealNetwork{})
                if err != nil {
                        klog.ErrorS(err, "Failed to get node ip address matching node port addresses, health check port will listen to all node addresses", "nodePortAddr
esses", nodePortAddresses)
-                       nodeIPs = []net.IP{net.IPv4zero}
+               } else {
+                       nodeIPs = ips
                }
        }
```